editor/src/components/editor/store/editor-dispatch-performance-logging.tsx

```diff
@@ -0,0 +1,63 @@
+import { PERFORMANCE_MARKS_ALLOWED } from '../../../common/env-vars'
+import { isFeatureEnabled } from '../../../utils/feature-switches'
+import { EditorAction } from '../action-types'
+import { simpleStringifyActions } from './dispatch'
+
+export function createPerformanceMeasure() {
+  const MeasureSelectorsEnabled = isFeatureEnabled('Debug – Measure Selectors')
+  const PerformanceMarks =
+    (isFeatureEnabled('Debug – Performance Marks (Slow)') ||
+      isFeatureEnabled('Debug – Performance Marks (Fast)')) &&
+    PERFORMANCE_MARKS_ALLOWED
+
+  let stringifiedActions = ''
+
+  let tasks: {
+    [name: string]: { timingInfo: string; startMarkName: string; endMarkName: string }
+  } = {}
+  return {
+    logActions(actions: ReadonlyArray<EditorAction>): void {
+      stringifiedActions = simpleStringifyActions(actions)
+    },
+    taskTime<T>(name: string, task: () => T): T {
```

Comment:
I think we should just early return in this function if `!MeasureSelectorsEnabled`, purely so that we don't later accidentally add extra computation that shouldn't be running when they're not enabled